build.gradle

```diff
@@ -120,7 +120,8 @@ project('common') {
             project(':shared:protocol').sourceSets.main.java
         }
         dependencies {
-            compile 'org.projectlombok:lombok:' + lombokVersion   
+            compile 'org.projectlombok:lombok:' + lombokVersion
+            compile group: 'io.netty', name: 'netty-all', version:  nettyVersion   
```

Comment:
This is probably a mistake. Anything referring to nettty should not live in common. Common should not be pravega specific. Whatever is causing the problem this is fixing is in error.

Reply:
the problem was with the javadoc source section in the common project javadoc section in build.gradle that cites:
 project(':shared:protocol').sourceSets.main.java

    This was causing these type of errors when building javadocs:
symbol:   class ByteBuf
  location: class CommandDecoder
/home/beth/fixBuild/pravega/shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandDecoder.java:46: error: cannot find symbol
    public static WireCommand parseCommand(ByteBuf in) throws IOException {
                                           ^
  symbol:   class ByteBuf
  location: class CommandDecoder
68 warnings


symbol:   class ByteBufInputStream
  location: class TableKeyDoesNotExist
/home/beth/fixBuild/pravega/shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java:1842: error: cannot find symbol
        public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOException {
If I take out the shared reference in sources, it seems to be ok. I'm not sure why it's there, so I just tried to make it happy.  I agree it would be best to not have the netty references escaping the shared/protocol area. Do you know the impact of taking out the shared/ protocol reference line in the java docs?  It seems like the javadocs for that is already generated in the shared/protocol area already. That said, I'm compiling with openjdk 8, which seems like it's less solid with 10. I can try compiling with 10 to see if the errors go away.. but I'm not hopeful about that. I'll see if I can find who added the source line in. Without that line common seems to javadoc cleanly without dragging netty into it.  So maybe the best answer is getting rid of it since it doesn't seem like it's required for javadoc in common.

Reply:
Removed a reference to shared source. Could then remove the netty reference without the errors occurring anymore.

Reply:
Tom advised it was probably an oversight when separating the two projects.